src/pybind/mgr/dashboard/plugins/feature_toggles.py

```diff
@@ -101,7 +101,7 @@ def cmd(mgr,
                             'enabled' if action == Actions.ENABLE.value else
                             'disabled')]
             else:
-                for feature in features or [f.value for f in Features]:
+                for feature in features or list(Features):
```

Comment:
Mmm I don't think this is equivalent:

```py
[<Features.RBD: 'rbd'>, <Features.MIRRORING: 'mirroring', ...] != ['rbd', 'mirroring', ...]
```

Reply:
1. `def cmd` takes the features as arguments: `features: Optional[List[Features]] = None`. Which means `features` in this line is of type `List[Features]`. 

2. In order for `features or [f.value for f in Features]` to make sense, we should make sure that the first part of `or` has the same type as the second part. `[f.value for f in Features]` is of type `List[str]`. So right now, the `feature` is either of type `Feature` or of type `str` based on the value of `features`. 

The current code only works, because `"{}".format(feature)` returns the same value for both `Feature` and `str`.

Reply:
I think part of the confusion here is that the `cmd(.., features: Optional[List[Features]], ...)` type hint is not correct:
* There's no way for a `MonCommand` to cast a CLI string to a `Feature` Enum: it'll always be a `str` (or a list of `str`s): "rbd,iscsi,rbdmirror".
* [The proper syntax](https://www.python.org/dev/peps/pep-0586/#interactions-with-enums-and-exhaustiveness-checks) here would have been `features: Optional[List[Literal[Features]]]`.

Reply:
Sure? @tchaikov added support to annotate CLI commands with Enums and then generate the ceph command description out of the python types. 

See 

https://github.com/ceph/ceph/blob/f744878d82d96c3fd9d11ef5bd74abf321a62b72/src/pybind/mgr/orchestrator/module.py#L35-L39

and 

https://github.com/ceph/ceph/blob/f744878d82d96c3fd9d11ef5bd74abf321a62b72/src/pybind/mgr/orchestrator/module.py#L347-L348

Reply:
Something is broken here.

Let's assume:
```py
class Features(Enum):
    RBD = 'rbd'
    RGW = 'rgw'
    ...
```

If `feature` is the `Feature.<enum>.name` item ('RBD', 'RGW', ...), then the following line is broken (as it should be 'rbd', 'rgw'):
```py
msg += ["Feature '{}': {}".format(
   feature
```

If `feature` is the `Feature.<enum>.value` item ('rbd', 'rgw', ...), then the following line is broken (as it should be 'RBD', 'RGW'):
```py
    enabled = mgr.get_module_option(self.OPTION_FMT.format(feature))
```